File: BOJ/2872/s2.js
const [n, ...arr] = require('fs')
  .readFileSync('input.txt')
  .toString()
  .trim()
  .split('\n') // \r\n으로 잘라서 계속 틀렸다...
  .map((v) => +v);

const lastBookIdx = arr.indexOf(n);
let bookNum = n - 1;
let fixedBooks = 1;

for (let i = lastBookIdx - 1; i >= 0; i--) {
  if (arr[i] === bookNum) {
    bookNum--;
    fixedBooks++;
  }
}

console.log(arr.length - fixedBooks);

File: BOJ/15651/s2.py
# 정답

N, M = map(int, input().split())

arr = [0] * M

def DFS(curr):
    if curr == M:
        print(' '.join(arr))
        return
    else:
        for i in range(1, N+1):
            arr[curr] = str(i)
            DFS(curr + 1)

DFS(0)